backend/app/routers/experiment_templates.py

```diff
@@ -93,6 +120,74 @@ async def create_experiment_template(
     return experiment_template.map_to_response()
 
 
+@router.put("/experiment-templates/{id}", response_model=ExperimentTemplateResponse)
+async def update_experiment_template(
+    id: PydanticObjectId,
+    experiment_template_req: ExperimentTemplateCreate,
+    user: dict = Depends(get_current_user(required=True)),
+) -> Any:
+    old_exp_template = await get_experiment_template_if_accessible_or_raise(
+        id, user, write_access=True
+    )
+    exist_experiments = (
```

Comment:
I recommend either moving these checks (experiments and others_experiments) to the inner function or renaming the arguments to what it actually does. In this way some pre computation is needed before the function call and it is also not clear why the function needs this. Something like `update_visibility: bool`.

Reply:
I was trying not to perform any DB operations within the `app/models/` directory as it is my understanding that these classes representing individual models should be rather light and only perform operations on its own object if possible. That was why the checks for existence of experiments corresponding to a specific template were realized in the endpoint itself which isn't ideal either.

I'm wondering whether my idea of what a class representing a document within a DB should look like is flawed or not, but I imagine, ideally we would create a new separate service that performs various validation checks on database documents and the classes in models directory inheriting the beanie.Document class would be freed of performing async calls to database.

In any case, for now, I will put additional calls to the DB to the `update_template` function, and later on we can discuss the idea of extracting such DB calls into a separate new service. This would also mean moving of ExperimentTemplate functions like `validate_datasets, validate_models, ...` into that newly created service as well, to a certain degree.